File: csharp/beer-song/BeerSong.cs
using System;
using System.Text;

public static class BeerSong
{
    public static string Recite(int startBottles, int takeDown)
    {
        StringBuilder s = new StringBuilder();
        for (int i = 0; i < takeDown; i++, startBottles--) {
            string startBottles1 = startBottles == 0 ? "No more" : startBottles.ToString();
            string startBottles2 = startBottles == 0 ? "no more" : startBottles.ToString();
            string bottleString = $"bottle{(startBottles == 1 ? "" : "s")}";
            string oneOrIt = $"{(startBottles > 1 ? "one" : "it")}";
            string remainingBottles = $"{(startBottles > 1 ? (startBottles - 1).ToString() : "no more")} bottle{(startBottles == 2 ? "" : "s")}";

            s.Append($"{startBottles1} {bottleString} of beer on the wall, {startBottles2} {bottleString} of beer.\n");
            if (startBottles == 0) {
                s.Append("Go to the store and buy some more, 99 bottles of beer on the wall.");
            } else {
                s.Append($"Take {oneOrIt} down and pass it around, {remainingBottles} of beer on the wall.");
            }
            if (i < takeDown - 1) {
                s.Append("\n\n");
            }
        }
        return s.ToString();
    }
}

File: swift/beer-song/Sources/BeerSong/BeerSong.swift
struct BeerSong {
    let n: Int
    
    init(numberOfBeerBottles: Int) {
        n = numberOfBeerBottles
    }
    
    func generateVersesOfBeerSong() -> String {
        var res = [String]()
        var i = n
        while i >= 0 {
            let numBottles1 = i == 0 ? "No more" : String(i)
            let numBottles2 = i == 0 ? "no more" : String(i)
            let bottles = i == 1 ? "bottle" : "bottles"
            let bottles2 = i == 2 ? "bottle" : "bottles"
            let remaining = i == 1 ? "no more" : String(i - 1)
            
            res.append("\(numBottles1) \(bottles) of beer on the wall, \(numBottles2) \(bottles) of beer.")
            if (i == 0) {
                res.append("Go to the store and buy some more, 99 bottles of beer on the wall.")
            } else {
                res.append("Take one down and pass it around, \(remaining) \(bottles2) of beer on the wall.\n")
            }
            i -= 1
        }
        return res.joined(separator: "\n")
    }
}
